customization-base/src/main/java/com/azure/autorest/customization/ClassCustomization.java

```diff
@@ -87,6 +87,51 @@ public ClassCustomization addImports(String... imports) {
         return this;
     }
 
+    /**
+     * Adds a static block to the class. The {@code staticCodeBlock} should include the static keyword followed by
+     * the static code.
+     * @param staticCodeBlock The static code block including the static keyword.
+     * @return The updated {@link ClassCustomization}.
+     */
+    public ClassCustomization addStaticBlock(String staticCodeBlock) {
+        return addStaticBlock(staticCodeBlock, null);
+    }
+
+    /**
+     * Adds a static block to the class. The {@code staticCodeBlock} should include the static keyword followed by
+     * the static code.
+     * @param staticCodeBlock The static code block including the static keyword.
+     * @param importsToAdd The list of imports to add to the class.
+     * @return The updated {@link ClassCustomization}.
+     */
+    public ClassCustomization addStaticBlock(String staticCodeBlock, List<String> importsToAdd) {
+        int staticBlockStartLine;
+        // Find all constructor and field symbols.
+        List<SymbolInformation> staticBlockLocationFinder = languageClient.listDocumentSymbols(fileUri).stream()
+                .filter(symbol -> symbol.getKind() == SymbolKind.FIELD)
+                .collect(Collectors.toList());
+
+        if (Utils.isNullOrEmpty(staticBlockLocationFinder)) {
+            staticBlockStartLine = symbol.getLocation().getRange().getStart().getLine();
+        } else {
+            SymbolInformation symbol = staticBlockLocationFinder.get(staticBlockLocationFinder.size() - 1);
+            staticBlockStartLine = symbol.getLocation().getRange().getStart().getLine();
+        }
+        int indentAmount = Utils.getIndent(editor.getFileLine(fileName, staticBlockStartLine)).length();
```

Comment:
Couldn't this produce the wrong indentation level if a field isn't found and the class symbol is used to determine the static block start line?

Reply:
Customizations shouldn't really include indentation logic ideally. But this pattern is used in all customizations. The formatter at the end of the code generation should ensure everything is indented correctly.